### PR TITLE
Better documentation

### PR DESCRIPTION
**Pull request by londumas**
Here is a commit at a proposition of better documentation.
My goal is for us to agree on one file and then go very fast on the rest of them.
Please @ngbusca and @mblomqvist, tell me what you think about name of variables
and comments.
Step toward fixing https://github.com/igmhub/picca/issues/434

### CODE REVIEW

File: bin/do_xcf.py [hunk 1]
```diff
@@ -9,83 +9,97 @@
...
-    tmp = xcf.xcf(p)
-    return tmp
+def corr_func(pixels):
+    """Send correlation on one processor for a list of healpix
+    
+    Args:
+        pixels (list of int): list of healpix to compute
+            the correlation on.
+
+    Returs:
```
Comment (mblomqvist) [inline bin/do_xcf.py]: Returns:

File: bin/do_xcf.py [hunk 2]
```diff
@@ -9,83 +9,97 @@
...
-                    help = 'number of processors')
+    parser.add_argument('--z-cut-min', type = float, default=0., required=False,
+        help = 'Use only pairs of forest x object with the mean of the last absorber \
+        redshift and the object redshift higher than z-cut-min')
 
-    parser.add_argument('--z-min-obj', type = float, default = None, required=False,
-                    help = 'min redshift for object field')
+    parser.add_argument('--z-cut-max', type=float, default=10., required=False,
+        help = 'Use only pairs of forestx object with the mean of the last absorber \
+        redshift and the object redshift smaller than z-cut-max')
```
Comment (mblomqvist) [inline bin/do_xcf.py]: forest x object

File: bin/do_xcf.py [hunk 3]
```diff
@@ -9,83 +9,97 @@
...
-    parser.add_argument('--nside', type = int, default = 16, required=False,
-                    help = 'healpix nside')
+    parser.add_argument('--z-max-obj', type=float, default=None, required=False,
+        help = 'Max redshift for object field')
 
-    parser.add_argument('--nproc', type = int, default = None, required=False,
-                    help = 'number of processors')
+    parser.add_argument('--z-cut-min', type = float, default=0., required=False,
+        help = 'Use only pairs of forest x object with the mean of the last absorber \
+        redshift and the object redshift higher than z-cut-min')
```
Comment (mblomqvist) [inline bin/do_xcf.py]: larger than

File: bin/do_xcf.py [hunk 4]
```diff
@@ -9,83 +9,97 @@
...
 
-    parser.add_argument('--no-remove-mean-lambda-obs', action="store_true", required=False,
-                    help = 'Do not remove mean delta versus lambda_obs')
+    parser.add_argument('--nproc', type=int, default=None, required=False,
+        help = 'Number of processors')
 
-    parser.add_argument('--from-image', type = str, default = None, required=False,
-                    help = 'use image format to read deltas', nargs='*')
+    parser.add_argument('--nspec', type=int, default=None, required=False,
+        help = 'Maximum spectra to read')
```
Comment (mblomqvist) [inline bin/do_xcf.py]: Maximum number of

File: bin/do_xcf.py [hunk 5]
```diff
@@ -186,17 +200,22 @@ def corr_func(p):
...
-    head['NSIDE']=xcf.nside
-
-    out.write([rp,rt,z,nb],names=['RP','RT','Z','NB'],header=head)
-    head2 = [{'name':'HLPXSCHM','value':'RING','comment':'healpix scheme'}]
-    out.write([hep,wes,cfs],names=['HEALPID','WE','DA'],header=head2)
+    head = [ {'name':'RPMIN','value':xcf.rp_min,'comment':'Minimum r-parallel'},
+        {'name':'RPMAX','value':xcf.rp_max,'comment':'Maximum r-parallel'},
+        {'name':'RTMAX','value':xcf.rt_max,'comment':'Maximum r-transverse'},
+        {'name':'NP','value':xcf.np,'comment':'Number bin r-parallel'},
+        {'name':'NT','value':xcf.nt,'comment':'Number bin r-transverse'},
```
Comment (mblomqvist) [inline bin/do_xcf.py]: Number of bins (if that many characters are allowed in a header)

File: bin/do_xcf.py [hunk 6]
```diff
@@ -186,17 +200,22 @@ def corr_func(p):
...
+        {'name':'Z_CUT_MIN','value':xcf.z_cut_min,'comment':'Minimum redshift of pairs'},
```
Comment (mblomqvist) [inline bin/do_xcf.py]: Minimum redshift of pairs is not strictly true, but I have no better suggestion.

File: bin/do_xcf.py [hunk 7]
```diff
@@ -186,17 +200,22 @@ def corr_func(p):
...
+        {'name':'Z_CUT_MAX','value':xcf.z_cut_max,'comment':'Maximum redshift of pairs'},
+        {'name':'NSIDE','value':xcf.nside,'comment':'Healpix nside'}
+    ]
+    out.write([rp,rt,z,nb],names=['RP','RT','Z','NB'],
+        comment=['R-parallel','R-transverse','Redshift','Number of pairs'],
+        header=head,extname='ATTRIBUTES')
+
+    head2 = [{'name':'HLPXSCHM','value':'RING','comment':'Healpix scheme'}]
```
Comment (mblomqvist) [inline bin/do_xcf.py]: Separate comments for HLPXSCHM and RING?

Comment (mblomqvist) [reply bin/do_xcf.py]: Sorry, I read that wrong.

File: bin/do_xcf.py (hunk 6 above)
Comment (londumas) [reply bin/do_xcf.py]: indeed.

**Comment by londumas** [comment]
Thank you for the review @mblomqvist.

File: bin/do_xcf.py [hunk 8]
```diff
@@ -186,17 +200,22 @@ def corr_func(p):
...
-    out.write([rp,rt,z,nb],names=['RP','RT','Z','NB'],header=head)
-    head2 = [{'name':'HLPXSCHM','value':'RING','comment':'healpix scheme'}]
-    out.write([hep,wes,cfs],names=['HEALPID','WE','DA'],header=head2)
+    head = [ {'name':'RPMIN','value':xcf.rp_min,'comment':'Minimum r-parallel'},
+        {'name':'RPMAX','value':xcf.rp_max,'comment':'Maximum r-parallel'},
+        {'name':'RTMAX','value':xcf.rt_max,'comment':'Maximum r-transverse'},
+        {'name':'NP','value':xcf.np,'comment':'Number of bins in r-parallel'},
+        {'name':'NT','value':xcf.nt,'comment':'Number of bins in r-transverse'},
+        {'name':'Z_CUT_MIN','value':xcf.z_cut_min,'comment':'Minimum redshift of pairs'},
+        {'name':'Z_CUT_MAX','value':xcf.z_cut_max,'comment':'Maximum redshift of pairs'},
```
Comment (ngbusca) [inline bin/do_xcf.py]: @londumas could you change `Z_CUT_MIN` and `Z_CUT_MAX` to `ZCUTMIN` and `ZCUTMAX` ? keyword names in .fits must be 8 characters or less...

**Comment by londumas** [comment]
@ngbusca, @mblomqvist, @vserret, would you accept to split the work on adding documentation?
I could take care of all the files in `bin/` and also in `py/picca/test/`. We have a growing number of user and it would be nice to have the documentation written. You would just need to add a preamble like the following one to all functions and methods and add a comment to all variables written FITS file and add a `EXTNAME`.
```
    """Send correlation on one processor for a list of healpix
    
    Args:
        pixels (list of int): list of healpix to compute
            the correlation on.

    Returns:
        cor (list of scipy array): list of array with the
            computed correlation and other attributes.
    
    """

```

**Comment by ngbusca** [comment]
I think this effort would be much more useful if directed towards filling readthedocs. It's ok to have good command help messages but I doubt someone is going to learn to run picca by reading them.

**Comment by londumas** [comment]
I though that readthedocs was using automatically the documentation inside the code. Is it not?
I totally agree that documentation of functions is not that useful for end-user, however I do believe that:
- documentation, i.e. comments, inside produced FITS file is very important.
- documentation of our `do_*.py` scripts is very useful for end-users.
What do you think @ngbusca and @mblomqvist ?

**Comment by mblomqvist** [comment]
@londumas I agree that comments in FITS files is useful and that the help messages in the do_*.py scripts may need some updating. For example, the 'mode' option of do_deltas.py doesn't mention that one can run on spPlates.

**Comment by ngbusca** [comment]
@londumas tell me when it's ready

**Comment by londumas** [comment]
@ngbusca, it is ready.
Unitest are very useful !

**Pull request merged**